libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationTelemetry.java

```diff
@@ -157,44 +156,33 @@ void initializeLifecycleMonitor(Application application) {
    * @param directionsRoute first route passed to navigation
    */
   void startSession(DirectionsRoute directionsRoute) {
-    if (!isConfigurationChange) {
-      navigationSessionState = navigationSessionState.toBuilder()
-        .sessionIdentifier(TelemetryUtils.obtainUniversalUniqueIdentifier())
-        .tripIdentifier(TelemetryUtils.obtainUniversalUniqueIdentifier())
-        .originalDirectionRoute(directionsRoute)
-        .originalRequestIdentifier(directionsRoute.routeOptions().requestUuid())
-        .requestIdentifier(directionsRoute.routeOptions().requestUuid())
-        .currentDirectionRoute(directionsRoute)
-        .eventRouteDistanceCompleted(0)
-        .mockLocation(metricLocation.getLocation().getProvider().equals(MOCK_PROVIDER))
-        .rerouteCount(0)
-        .build();
-    }
-    isConfigurationChange = false;
+    navigationSessionState = navigationSessionState.toBuilder()
+      .sessionIdentifier(TelemetryUtils.obtainUniversalUniqueIdentifier())
+      .tripIdentifier(TelemetryUtils.obtainUniversalUniqueIdentifier())
+      .originalDirectionRoute(directionsRoute)
+      .originalRequestIdentifier(directionsRoute.routeOptions().requestUuid())
+      .requestIdentifier(directionsRoute.routeOptions().requestUuid())
+      .currentDirectionRoute(directionsRoute)
+      .eventRouteDistanceCompleted(0)
+      .mockLocation(metricLocation.getLocation().getProvider().equals(MOCK_PROVIDER))
+      .rerouteCount(0)
+      .build();
     sendRouteRetrievalEventIfExists();
     fireOffBatteryScheduler();
     gpsEventFactory.navigationStarted(navigationSessionState.sessionIdentifier());
   }
 
   void stopSession() {
-    if (navigationSessionState.startTimestamp() != null) {
-      NavigationMetricsWrapper.cancelEvent(
-        navigationSessionState, metricProgress, metricLocation.getLocation(), context
-      );
-    }
+    sendCancelEvent();
     gpsEventFactory.reset();
     resetDepartFactory();
   }
 
-  void endSession(boolean isConfigurationChange) {
-    this.isConfigurationChange = isConfigurationChange;
-    if (!isConfigurationChange) {
-      flushEventQueues();
-      lifecycleMonitor = null;
-      NavigationMetricsWrapper.disable();
-      isInitialized = false;
-    }
-    cancelBatteryScheduler();
```

Comment:
It looks like this was accidentally removed ☝️

Reply:
Good 👀 Thanks for the heads up!

Reply:
Updated @danesfeder